### PR TITLE
Enable IDE0075: SimplifyConditionalExpression

### PR DESCRIPTION
**Pull request by xtqqczze**
https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0075

### CODE REVIEW

File: src/System.Management.Automation/engine/hostifaces/InternalHostUserInterface.cs [hunk 1]
```diff
@@ -78,7 +78,7 @@ public override
 
         public override bool SupportsVirtualTerminal
         {
-            get { return (_externalUI != null) ? _externalUI.SupportsVirtualTerminal : false; }
+            get { return (_externalUI != null) && _externalUI.SupportsVirtualTerminal; }
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/hostifaces/InternalHostUserInterface.cs:81]:
```suggestion
            get { return _externalUI != null && _externalUI.SupportsVirtualTerminal; }
```

File: src/System.Management.Automation/engine/hostifaces/PowerShell.cs [hunk 2]
```diff
@@ -5546,7 +5546,7 @@ internal bool ConstructPipelineAndDoWork(Runspace rs, bool performSyncInvoke)
                         LocalPipeline localPipeline = new LocalPipeline(
                             lrs,
                             _shell.Commands.Commands,
-                            ((_settings != null) && (_settings.AddToHistory)) ? true : false,
+                            ((_settings != null) && _settings.AddToHistory),
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/hostifaces/PowerShell.cs:5549]:
```suggestion
                            (_settings != null && _settings.AddToHistory),
```

File: src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs [hunk 3]
```diff
@@ -98,8 +98,7 @@ private ClientMethodExecutor(BaseClientTransportManager transportManager, PSHost
                 if (hostPrivateData != null)
                 {
                     PSNoteProperty allowSetShouldExit = hostPrivateData.Properties["AllowSetShouldExitFromRemote"] as PSNoteProperty;
-                    hostAllowSetShouldExit = (allowSetShouldExit != null && allowSetShouldExit.Value is bool) ?
-                        (bool)allowSetShouldExit.Value : false;
+                    hostAllowSetShouldExit = (allowSetShouldExit != null && allowSetShouldExit.Value is bool) && (bool)allowSetShouldExit.Value;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs:101]:
```suggestion
                    hostAllowSetShouldExit = allowSetShouldExit != null && allowSetShouldExit.Value is bool && (bool)allowSetShouldExit.Value;
```

File: src/System.Management.Automation/engine/remoting/client/Job.cs [hunk 4]
```diff
@@ -2968,7 +2968,7 @@ internal override bool CanDisconnect
             get
             {
                 RemoteRunspace remoteRS = Runspace as RemoteRunspace;
-                return (remoteRS != null) ? remoteRS.CanDisconnect : false;
+                return (remoteRS != null) && remoteRS.CanDisconnect;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/client/Job.cs:2971]:
```suggestion
                return remoteRS != null && remoteRS.CanDisconnect;
```

File: src/System.Management.Automation/engine/remoting/client/RemotingProtocol2.cs [hunk 5]
```diff
@@ -233,7 +233,7 @@ internal void CreatePowerShellOnServerAndInvoke(ClientRemotePowerShell shell)
 
             // Find out if this is an invoke and disconnect operation and if so whether the endpoint
             // supports disconnect.  Throw exception if disconnect is not supported.
-            bool invokeAndDisconnect = (shell.Settings != null) ? shell.Settings.InvokeAndDisconnect : false;
+            bool invokeAndDisconnect = (shell.Settings != null) && shell.Settings.InvokeAndDisconnect;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/client/RemotingProtocol2.cs:236]:
```suggestion
            bool invokeAndDisconnect = shell.Settings != null && shell.Settings.InvokeAndDisconnect;
```

File: src/System.Management.Automation/engine/remoting/client/RemotingProtocol2.cs [hunk 6]
```diff
@@ -938,7 +938,7 @@ internal bool EndpointSupportsDisconnect
             get
             {
                 WSManClientSessionTransportManager wsmanTransportManager = _transportManager as WSManClientSessionTransportManager;
-                return (wsmanTransportManager != null) ? wsmanTransportManager.SupportsDisconnect : false;
+                return (wsmanTransportManager != null) && wsmanTransportManager.SupportsDisconnect;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/client/RemotingProtocol2.cs:941]:
```suggestion
                return wsmanTransportManager != null && wsmanTransportManager.SupportsDisconnect;
```

File: src/System.Management.Automation/engine/remoting/client/RunspaceRef.cs [hunk 7]
```diff
@@ -93,7 +93,7 @@ private PSCommand ParsePsCommandUsingScriptBlock(string line, bool? useLocalScop
                 // and if we are not in a loopback configuration mode, in which case we always force remote script commands
                 // to be parsed and evaluated on the remote session (not in the current local session).
                 RemoteRunspace remoteRunspace = _runspaceRef.Value as RemoteRunspace;
-                bool isConfiguredLoopback = (remoteRunspace != null) ? remoteRunspace.IsConfiguredLoopBack : false;
+                bool isConfiguredLoopback = (remoteRunspace != null) && remoteRunspace.IsConfiguredLoopBack;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/client/RunspaceRef.cs:96]:
```suggestion
                bool isConfiguredLoopback = remoteRunspace != null && remoteRunspace.IsConfiguredLoopBack;
```

File: src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs [hunk 8]
```diff
@@ -3003,9 +3003,9 @@ public virtual string[] ComputerName
                     var matchingRunspaceInfos = remoteRunspaceInfos
                         .Where<PSSession>(session => (supportWildChar ? inputNamePattern.IsMatch(session.VMName)
                                                                       : inputName.Equals(session.ContainerId)) &&
-                                                     ((sessionNamePattern == null) ? true : sessionNamePattern.IsMatch(session.Name)) &&
+                                                     ((sessionNamePattern == null) || sessionNamePattern.IsMatch(session.Name)) &&
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs:3006]:
```suggestion
                                                     (sessionNamePattern == null || sessionNamePattern.IsMatch(session.Name)) &&
```

File: src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs [hunk 9]
```diff
@@ -3003,9 +3003,9 @@ public virtual string[] ComputerName
...
                                                      QueryRunspaces.TestRunspaceState(session.Runspace, filterState) &&
-                                                     ((configurationNamePattern == null) ? true : configurationNamePattern.IsMatch(session.ConfigurationName)) &&
+                                                     ((configurationNamePattern == null) || configurationNamePattern.IsMatch(session.ConfigurationName)) &&
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs:3008]:
```suggestion
                                                     (configurationNamePattern == null || configurationNamePattern.IsMatch(session.ConfigurationName)) &&
```

File: src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs [hunk 10]
```diff
@@ -3063,7 +3063,7 @@ public virtual string[] ComputerName
                                                                       : inputName.Equals(session.ContainerId)) &&
                                                      sessionInstanceId.Equals(session.InstanceId) &&
                                                      QueryRunspaces.TestRunspaceState(session.Runspace, filterState) &&
-                                                     ((configurationNamePattern == null) ? true : configurationNamePattern.IsMatch(session.ConfigurationName)) &&
+                                                     ((configurationNamePattern == null) || configurationNamePattern.IsMatch(session.ConfigurationName)) &&
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs:3066]:
```suggestion
                                                     (configurationNamePattern == null || configurationNamePattern.IsMatch(session.ConfigurationName)) &&
```

File: src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs [hunk 11]
```diff
@@ -3106,9 +3106,9 @@ public virtual string[] ComputerName
 
                     var matchingRunspaceInfos = remoteRunspaceInfos
                         .Where<PSSession>(session => vmId.Equals(session.VMId) &&
-                                                     ((sessionNamePattern == null) ? true : sessionNamePattern.IsMatch(session.Name)) &&
+                                                     ((sessionNamePattern == null) || sessionNamePattern.IsMatch(session.Name)) &&
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs:3109]:
```suggestion
                                                     (sessionNamePattern == null || sessionNamePattern.IsMatch(session.Name)) &&
```

File: src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs [hunk 12]
```diff
@@ -3106,9 +3106,9 @@ public virtual string[] ComputerName
...
                                                      QueryRunspaces.TestRunspaceState(session.Runspace, filterState) &&
-                                                     ((configurationNamePattern == null) ? true : configurationNamePattern.IsMatch(session.ConfigurationName)) &&
+                                                     ((configurationNamePattern == null) || configurationNamePattern.IsMatch(session.ConfigurationName)) &&
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs:3111]:
```suggestion
                                                     (configurationNamePattern == null || configurationNamePattern.IsMatch(session.ConfigurationName)) &&
```

File: src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs [hunk 13]
```diff
@@ -3143,7 +3143,7 @@ public virtual string[] ComputerName
                         .Where<PSSession>(session => vmId.Equals(session.VMId) &&
                                                      sessionInstanceId.Equals(session.InstanceId) &&
                                                      QueryRunspaces.TestRunspaceState(session.Runspace, filterState) &&
-                                                     ((configurationNamePattern == null) ? true : configurationNamePattern.IsMatch(session.ConfigurationName)) &&
+                                                     ((configurationNamePattern == null) || configurationNamePattern.IsMatch(session.ConfigurationName)) &&
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs:3146]:
```suggestion
                                                     (configurationNamePattern == null || configurationNamePattern.IsMatch(session.ConfigurationName)) &&
```

File: src/System.Management.Automation/engine/remoting/server/ServerRemoteHost.cs [hunk 14]
```diff
@@ -335,7 +335,7 @@ public override void PushRunspace(Runspace runspace)
             // PSEdit support.  Existence of RemoteSessionOpenFileEvent event indicates host supports PSEdit
             _hostSupportsPSEdit = false;
             PSEventManager localEventManager = Runspace?.Events;
-            _hostSupportsPSEdit = (localEventManager != null) ? localEventManager.GetEventSubscribers(HostUtilities.RemoteSessionOpenFileEvent).GetEnumerator().MoveNext() : false;
+            _hostSupportsPSEdit = (localEventManager != null) && localEventManager.GetEventSubscribers(HostUtilities.RemoteSessionOpenFileEvent).GetEnumerator().MoveNext();
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/server/ServerRemoteHost.cs:338]:
```suggestion
            _hostSupportsPSEdit = localEventManager != null && localEventManager.GetEventSubscribers(HostUtilities.RemoteSessionOpenFileEvent).GetEnumerator().MoveNext();
```

File: src/System.Management.Automation/FormatAndOutput/common/FormattingObjectsDeserializer.cs [hunk 15]
```diff
@@ -290,7 +290,7 @@ internal int DeserializeIntMemberVariable(PSObject so, string property)
         internal bool DeserializeBoolMemberVariable(PSObject so, string property, bool cannotBeNull = true)
         {
             var val = DeserializeMemberVariable(so, property, typeof(bool), cannotBeNull);
-            return (val == null) ? false : (bool)val;
+            return val != null && (bool)val;
```
[the remark this answers is not in the document]
Comment (xtqqczze) [reply src/System.Management.Automation/FormatAndOutput/common/FormattingObjectsDeserializer.cs:293]: The `as` operator cannot be used with a value type.

[the remark this answers is not in the document]
Comment (xtqqczze) [reply src/System.Management.Automation/FormatAndOutput/common/FormattingObjectsDeserializer.cs:293]: We can do `return val as bool? ?? false` but I don't think it is an improvement.

[the remark this answers is not in the document]
Comment (xtqqczze) [reply src/System.Management.Automation/FormatAndOutput/common/FormattingObjectsDeserializer.cs:293]: > We can do `return val as bool? ?? false` but I don't think it is an improvement.

It's equivalent to `return (val as bool?).GetValueOrDefault()`

File: src/System.Management.Automation/engine/GetCommandCommand.cs [hunk 16]
```diff
@@ -1632,8 +1632,7 @@ private static PSObject GetParameterType(Type parameterType)
                 new ArrayList(Enum.GetValues(parameterType)) : new ArrayList();
             returnParameterType.Properties.Add(new PSNoteProperty("EnumValues", enumValues));
 
-            bool hasFlagAttribute = (isArray) ?
-                ((parameterType.GetCustomAttributes(typeof(FlagsAttribute), true)).Length > 0) : false;
+            bool hasFlagAttribute = (isArray) && ((parameterType.GetCustomAttributes(typeof(FlagsAttribute), true)).Length > 0);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/GetCommandCommand.cs:1635]:
```suggestion
            bool hasFlagAttribute = isArray && ((parameterType.GetCustomAttributes(typeof(FlagsAttribute), true)).Length > 0);
```

Comment (xtqqczze) [reply src/System.Management.Automation/engine/GetCommandCommand.cs:1635]: We can remove more redundant parenthesis in new PR.

Comment (xtqqczze) [reply src/System.Management.Automation/engine/GetCommandCommand.cs:1635]: I mean we can use an automated code fix, rather than picking changes manually.

**Comment by xtqqczze** [comment]
@iSazonov Ae you happy to merge as-if, or do you want more changes to remove redundant parenthesis?

**Review by iSazonov — APPROVED** [review]

**Pull request merged**